### PR TITLE
Upgrade to dat-node@1. Fixes #168

### LINKED ISSUE #168: upgrade to dat-node@latest
for #167

### PR DESCRIPTION
**Pull request by juliangruber**
Major changes:

- moved from emitting events on the dat instance, which isn't an eventemitter any more, to an emitter at `dat.events`
- no need to persist `dat.owner` any more
- make `dat.stats` a getter to the underlying hyperstats instance
- no need to manually close `dat.db()` any more

### CODE REVIEW

File: lib/dat-manager.js [hunk 1]
```diff
@@ -19,18 +18,22 @@ module.exports = class Manager extends EventEmitter {
...
   }
   create (dir, cb) {
-    const dat = Dat({ dir })
-    dat.owner = true
-    dat.dir = dir
-    dat.share(err => {
+    Dat(dir, (err, dat) => {
       if (err) return cb(err)
+
+      dat.dir = dir
```
Comment (joehand) [inline lib/dat-manager.js]: Oops sorry changed API there. I can expose this if that'd be helpful. `dat.path` is the full resolved path but `dat.dir` could be the relative `dir` passed in.

File: lib/dat-manager.js [hunk 2]
```diff
@@ -44,102 +47,102 @@ module.exports = class Manager extends EventEmitter {
...
-          }
-          dat.title = json.title
-          dat.author = json.author
-          dat.emit('update')
-        })
+    dat.events = new EventEmitter()
+    dat.listStream = dat.archive.list({ live: true })
+    dat.listStream.on('data', entry => {
+      if (entry.name !== 'dat.json') return
+      collect(dat.archive.createFileReadStream('dat.json'), (err, raw) => {
```
Comment (joehand) [inline lib/dat-manager.js]: Ooh nice. So if there is an update to `dat.json` (i.e. more than one `dat.json` entry in metadata), the UI will just keep updating until the last metadata entry?

File: lib/dat-manager.js (hunk 1 above)
Comment (juliangruber) [inline lib/dat-manager.js]: oh, i wasn't aware of `dat.path`. I was adding `dat.dir` manually before I think, I can just use `dat.path` then

Comment (joehand) [inline lib/dat-manager.js]: 👍  the only difference is `dat.path` will be fully resolved, not sure if `dir` was there.

Comment (juliangruber) [inline lib/dat-manager.js]: addressed

File: lib/dat-manager.js (hunk 2 above)
Comment (juliangruber) [inline lib/dat-manager.js]: yup currently it's doing redundant work. Could we make it so the "live" stream only starts at the current point in time and goes forward?

**Comment by juliangruber** [comment]
thanks for the review so far, @joehand!

File: lib/dat-manager.js (hunk 2 above)
Comment (joehand) [inline lib/dat-manager.js]: No just wanted to confirm the UI would update. If it starts at current point you may not get the `dat.json`, right? (if its the first entry) Or maybe it'd be persisted in the db already?

Getting one file will be much easier with the append-tree update from mafintosh - so this probably is good for now.

**Comment by joehand** [comment]
Looks good! Hopefully it was relatively seamless to use the new dat-node 😬

File: lib/dat-manager.js (hunk 2 above)
Comment (juliangruber) [inline lib/dat-manager.js]: oh, yeah the ui will update, that's the `this.emit('update')` call underneath

**Comment by juliangruber** [comment]
@joehand after the versioning confusing yes it was :)

I would just also keep master the current stable version, otherwise it's confusing to look at the repo's code and see a different version from what you're using.

**Comment by yoshuawuyts** [comment]
We found multiple regressions, the first was with importing a dat and then reopening the window it would crash because `.dir` was renamed to `.path`. We've fixed this with the changes we've pushed

The second issue is that we cannot delete dats that we create and the window freezes without throwing an error. We created a dat that was ~50mb. We weren't able to fix this but tracked it down to the `remove` method in `data-manager`. `dat.close()` never calls a callback (cc/ @joehand)

screen shot 2017-01-27 at 12 45 25

**Comment by juliangruber** [comment]
> We found multiple regressions, the first was with importing a dat and then reopening the window it would crash because .dir was renamed to .path. We've fixed this with the changes we've pushed

Ah, thanks for catching this bug and fixing it!

> dat.close() never calls a callback (cc/ @joehand)

I've seen this too, in https://github.com/datproject/dat-node/issues/84. @joehand could you run the app and see if you can reproduce? Otherwise we could also try to create a node only gist of this happening.

File: elements/table.js [hunk 3]
```diff
@@ -195,15 +195,12 @@ function tableElement (dats, send) {
...
   return dats.map(dat => {
-    const stats = dat.hyperStats && dat.hyperStats.get()
-    const progress = stats
-      ? stats.blocksTotal
-        ? Math.round(stats.blocksProgress / stats.blocksTotal * 100)
-        : 100
-      : 0
+    const progress = dat.stats.blocksTotal
+      ? Math.round(dat.stats.blocksProgress / dat.stats.blocksTotal * 100)
+      : 100
```
Comment (juliangruber) [inline elements/table.js]: quick fix!

**Pull request merged**